sdk/tables/Azure.Data.Tables/src/TableEntity.cs

```diff
@@ -204,6 +204,7 @@ private protected void SetValue(string key, object value)
         /// <exception cref="InvalidOperationException">Value associated with given <paramref name="key"/> is not of type <paramref name="type"/>.</exception>
         private protected object GetValue(string key, Type type = null)
```

Comment:
Unrelated to this change, on line 227, we don't seem to be doing anything with the result of `IsAssignableFrom`

Reply:
oh oops hahaha I think this is from [this](https://github.com/Azure/azure-sdk-for-net/pull/13561#discussion_r458477747), and I must have just left it there because I wasn't sure whether to replace the if-expression.

I'll remove it for now.

Reply:
Just kidding, I used `IsAssignableFrom` to account for Nullable since the `Get*` methods return nullables for structs/value types. (for now)